### PR TITLE
Adding a route for Mandrill's url check.

### LINKED ISSUE #37609: Initial setup of Mandrill for Action Mailbox requires a specific token
### Steps to reproduce
Create a brand new rails install, and configure action mailbox to use mandrill.
Specify a secure mandrill_ingress_api_token, and deploy to a public URL. Mandrill will not be able to create the inbound email unless you configure your token to be "test-webhook".

### Expected behavior
I expect the process of creating a new mandrill inbound route from the mandrill api to be seamless. I should be able to create a new route and the rails app should respond to mandrill's test web hook successfully so that we can proceed to configure the inbound web hook.

### Actual behavior
Mandrill makes a request to the endpoint to validate that it a) exists, b) is secured with https, and b) that it validates signatures correctly. In order to do this, it signs this first request with the same token "test-webhook".

What I *think* should occur is that if we receive a web hook that validates against the token "test-webhook",  rails should consider it a no-op, and return 200, such that the inbound email can be created on the mandrill side.

### System configuration
Rails: 6.0.0
Ruby: 2.6.5

### PR DESCRIPTION
**Pull request by cromulus**
### Summary

Mandrill's Inbound API checks to see if a URL exists before it creates the web hook. It sends a HEAD request to  "/mandrill/inbound_emails", which rails converts into a GET request. This simply returns 200, and the word 'OK'.

Now we can generate inbound API calls with ease on Mandrill, without having to shuffle around tokens.

Fixes #37609
### Other Information

CHANGELOG
Simplifies the creation of Mandrill Inbound API urls, no longer have to set a temporary "test-webhook" token.

### CODE REVIEW

File: actionmailbox/test/controllers/ingresses/mandrill/inbound_emails_controller_test.rb
Comment (eugeneius) [inline actionmailbox/test/controllers/ingresses/mandrill/inbound_emails_controller_test.rb]: This should be `head`, right?

My reading of the documentation is that the HEAD request won't be signed at all - it's only the fallback POST that uses the "test-webhook" key.

Would you be able to verify this behaviour with a real Mandrill account?

Comment (cromulus) [reply actionmailbox/test/controllers/ingresses/mandrill/inbound_emails_controller_test.rb]: Indeed, looks like it. I'll make the change.

It also appears as if some of the tests in railties break as a result of this change:
`
railties/test/commands/routes_test.rb:166
railties/test/commands/routes_test.rb:202
activemodel/test/application/rake/routes_test.rb:20
` 

Should I update those tests so that they pass?

There are other tests that fail and I can't quite figure out why...
Should I continue to attempt to debug them?

Comment (eugeneius) [reply actionmailbox/test/controllers/ingresses/mandrill/inbound_emails_controller_test.rb]: Yes, please do update those tests to reflect the new route added here. Ideally those tests would be more resilient to incidental changes... but the easiest path here is to play along 🙂

The other failures are against the latest nightly snapshot of Ruby, and won't cause the built to fail on their own - you can see them also failing on the latest (passing) master build here:

https://buildkite.com/rails/rails/builds/67787

Comment (cromulus) [reply actionmailbox/test/controllers/ingresses/mandrill/inbound_emails_controller_test.rb]: Updated the tests. I *think* all of the tests that this change broke are now passing. Lets see what the buildkite comes back with. Regarding the conflicts below, I assume the maintainers can fix those problems?

File: actionmailbox/actionmailbox.gemspec
Comment (eugeneius) [inline actionmailbox/actionmailbox.gemspec:39]: Could you add this blank line back?

File: actionmailbox/config/routes.rb
Comment (eugeneius) [inline actionmailbox/config/routes.rb:7]: I'd add the new route above the other one and move both below this section, separated by a blank line, like so:

```ruby
    post "/sendgrid/inbound_emails" => "sendgrid/inbound_emails#create", as: :rails_sendgrid_inbound_emails

    # Mandrill checks for the existence of a URL with a HEAD request before it will create the webhook.
    get "/mandrill/inbound_emails"  => "mandrill/inbound_emails#health_check", as: :rails_mandrill_inbound_health_check
    post "/mandrill/inbound_emails" => "mandrill/inbound_emails#create",       as: :rails_mandrill_inbound_emails

    # Mailgun requires that a webhook's URL end in 'mime' for it to receive the raw contents of emails.
```

File: actionmailbox/test/controllers/ingresses/mandrill/inbound_emails_controller_test.rb
Comment (eugeneius) [inline actionmailbox/test/controllers/ingresses/mandrill/inbound_emails_controller_test.rb:13]:
```suggestion
  test "verifying existence of Mandrill inbound route" do
```

**Review by eugeneius — COMMENTED** [review]
This is looking good; I left a few style comments.

File: actionmailbox/test/controllers/ingresses/mandrill/inbound_emails_controller_test.rb
Comment (eugeneius) [reply actionmailbox/test/controllers/ingresses/mandrill/inbound_emails_controller_test.rb]: Looks like the tests are indeed passing now! ⭐️

We can fix the conflicts when merging if necessary, but if you're able to, rebasing your branch and fixing them yourself would help us out:

https://guides.rubyonrails.org/contributing_to_ruby_on_rails.html#update-your-branch

File: actionmailbox/config/routes.rb
Comment (cromulus) [reply actionmailbox/config/routes.rb:7]: 👍

File: actionmailbox/actionmailbox.gemspec
Comment (cromulus) [reply actionmailbox/actionmailbox.gemspec:39]: 👍

File: actionmailbox/config/routes.rb
Comment (cromulus) [reply actionmailbox/config/routes.rb]: Fixed in 8f107b0

File: actionmailbox/actionmailbox.gemspec
Comment (cromulus) [reply actionmailbox/actionmailbox.gemspec]: Fixed in 8f107b0

**Comment by cromulus** [comment]
@eugeneius  
I think we're ready to go!

**Comment by eugeneius** [comment]
I forgot to ask earlier: could you add an entry to the [changelog](https://github.com/rails/rails/blob/36b36fd57f5514e74d8b521c741d556a5a6a1251/actionmailbox/CHANGELOG.md)?

**Comment by cromulus** [comment]
Changelog updated. Thank you for all of your help with this!

**Comment by eugeneius** [comment]
I tweaked a few things on your branch and squashed to one commit again. This now looks good to me, but I'd like to see if George has any feedback before merging.

**Comment by cromulus** [comment]
Thanks!

**Review by georgeclaghorn — APPROVED** [review]

**Comment by eugeneius** [comment]
Backported to 6-0-stable in https://github.com/rails/rails/commit/5bf69b4bc550f42f547f70c8daffc0c126b8d1ad.

Thanks for sticking with this @cromulus 🙏

**Comment by cromulus** [comment]
Thanks for helping out! I'm thrilled to be able to contribute even in this small way.

**Pull request merged**
